### ISSUE TITLE
OpenFoodAPIClient.saveProduct fails to transfer nutriments

### ISSUE THREAD

**Issue by peterwvj** [openfoodfacts/openfoodfacts-dart#48]
I found what seems to be a critical bug in `OpenFoodAPIClient.saveProduct` that prevents nutriments from being saved.

### How to produce

Specify working user credentials and run the following code to add/edit a (real) product to the database:

```dart
import 'package:openfoodfacts/model/Nutriments.dart';
import 'package:openfoodfacts/model/Product.dart';
import 'package:openfoodfacts/model/Status.dart';
import 'package:openfoodfacts/model/User.dart';
import 'package:openfoodfacts/openfoodfacts.dart';

void main() async {
  // TODO: specify user credentials
  User myUser = User(userId: 'user', password: 'secret-password');

  var nutriments = Nutriments();
  nutriments.energy = 365;
  nutriments.carbohydrates = 12;
  nutriments.proteins = 6;
  nutriments.fat = 0.1;

  var newProduct = Product(
      barcode: "7340011364184",
      productName: "Chili beans",
      nutrimentDataPer: "100g",
      nutriments: nutriments);

  Status result = await OpenFoodAPIClient.saveProduct(myUser, newProduct);
  print(result.statusVerbose);
}
```
Next, look up the product by the barcode by entering https://world.openfoodfacts.org/api/v0/product/7340011364184.json into your browser.

### Expected result

I'd expect the values of the `code`,  `product_name`, `nutrition_data_per` and `nutriments` fields in `7340011364184.json` to match those in the code snippet above.

### Actual result

You'll see that the `code`, `product_name` and `nutrition_data_per` fields have the correct values but `nutriments` is empty, i.e. `"nutriments":{}`. So for some reason nutriments are not saved.

### Other information

Produced using version 0.3.10 of the API.

**Comment by peterwvj** [comment]
The body of the request seems to contain the nutriments information:

```
user_id=USERNAME&password=PASSWORD&code=7340011364184&product_name=Chili+beans&brands=null&brands_tags=null&lang=-&quantity=null&image_small_url=null&serving_size=null&serving_quantity=null&product_quantity=null&selected_images=%7B%7D&images=%7B%7D&nutriments=%7Bfat_100g%3A+0.1%2C+proteins_100g%3A+6.0%2C+energy_100g%3A+365.0%2C+carbohydrates_100g%3A+12.0%7D&additives_tags=null&allergens_tags=null&ingredients_analysis_tags=%5Ben%3Amaybe-vegan%2C+en%3Amaybe-vegetarian%2C+en%3Amaybe-palm-oil-free%5D&nutrition_data_per=100g
```

**Comment by peterwvj** [comment]
Stephane has informed me that the new version of the API requires each field be set individually, for example `https://world.openfoodfacts.org/cgi/product_jqm2.pl?code=7340011364184&product_name=Chili+beans&nutriment_fat=0.1&nutriment_fat_unit=g&nutriment_proteins=0.1&nutriment_proteins_unit=g`. So in short, the SDK is not updated to work with the new version of the PBI.

**Comment by peterwvj** [comment]
Here is a test that could be used to verify a fix:

```dart
    test('confirm that fields are saved', () async {
      double energy = 365;
      double carbs = 12;
      double proteins = 6;
      double fat = 0.1;
      var nutriments = Nutriments();
      nutriments.energy = energy;
      nutriments.carbohydrates = carbs;
      nutriments.proteins = proteins;
      nutriments.fat = fat;

      String barcode = "7340011364184";
      String productName = "Chili beans";
      String nutrimentDataPer = "100g";
      var newProduct = Product(
          barcode: barcode,
          productName: productName,
          nutrimentDataPer: nutrimentDataPer,
          nutriments: nutriments);

      Status status = await OpenFoodAPIClient.saveProduct(TestConstants.TEST_USER, newProduct);
    
      expect(status != null, true);
      expect(status.status, 1);
      expect(status.statusVerbose, "fields saved");
            
      ProductResult result = await OpenFoodAPIClient.getProductRaw(
          barcode, OpenFoodFactsLanguage.ENGLISH,
          user: TestConstants.TEST_USER);
            
      expect(result != null, true);
      expect(result.status, 1);
      expect(result.barcode, barcode);
      var searchedProduct = result.product;
      expect(searchedProduct != null, true);
      expect(searchedProduct.barcode, barcode);
      expect(searchedProduct.productName, productName);
      expect(searchedProduct.nutrimentDataPer, nutrimentDataPer);
      var searchedNutriments = searchedProduct.nutriments;
      expect(searchedNutriments.energy, energy);
      expect(searchedNutriments.carbohydrates, carbs);
      expect(searchedNutriments.proteins, proteins);
      expect(searchedNutriments.fat, fat);
    });
```

**Comment by monsieurtanuki** [comment]
Fixed by #187.
